src/modules/graphql/resolvers/class.resolver.ts

```diff
@@ -11,21 +14,46 @@ export class ClassResolver {
 
     @Query('classes')
     async getClasses(obj, args, context, info) {
-        return this.classPersistence.getAll();
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.CLASS);
+        const classes = await this.classPersistence.getAll();
+
+        if (permission === Permission.OWN) {
+            // find classes of teacher only
+            const teacher: UserDbModel = await this.userPersistenceService.getById(context.user.id);
+            const teacherClassId = teacher.class_id ? teacher.class_id.toString() : '';
+            return classes.filter((cls) => cls._id.toString() === teacherClassId);
+        }
+
+        return classes;
     }
 
     @Query('classById')
     async getClassById(obj, args, context, info) {
-        return this.classPersistence.getById(args.id);
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.CLASS);
+        const cls =  await this.classPersistence.getById(args.id);
+        if (permission === Permission.OWN) {
+            const user: UserDbModel = await this.userPersistenceService.getById(context.user.id);
+            const userClassId = user.class_id ? user.class_id.toString() : '';
+            return (cls._id.toString() === userClassId) ? cls : null;
+        }
+
+        return cls;
     }
 
     @Query('classByName')
     async getClassByName(obj, args, context, info) {
-        return this.classPersistence.getByName(args.name);
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.CLASS);
+        const cls = await this.classPersistence.getByName(args.name);
+        if (permission === Permission.OWN) {
+            const user: UserDbModel = await this.userPersistenceService.getById(context.user.id);
+            const userClassId = user.class_id ? user.class_id.toString() : '';
+            return (cls._id.toString() === userClassId) ? cls : null;
+        }
+        return cls;
```

Comment:
getClassByName and getClassById are almost the same except the argument delivered in the classPersistence. 
Can we refactor it to not include duplication of code?